clang/test/CodeGenSYCL/code_location.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -fsycl-int-header=%t.h %s -o %t.out
+// RUN: FileCheck -input-file=%t.h %s
+
+#include "Inputs/sycl.hpp"
+
+int main() {
+  cl::sycl::queue q;
+  q.submit([&](cl::sycl::handler &h) { h.single_task([] {}); });
+  q.submit([&](cl::sycl::handler &h) { h.single_task<class KernelName>([]() {}); });
```

Comment:
Please add more tests where kernel name is declared on different line/ kernel is defined as kernel object/etc